allow `FSharpTargetsPath` to be properly overwritten in a project file

brettfo opened the pull request:
Followup to #4922.

By previously setting `FSharpTargetsPath`, etc. directly we didn't allow for that to be manually set in a project, props, or target file.  By using this method the IDE can communicate where the compiler is located, but the user can still override this should they choose.

Edit:
With this change compiler redirection will require a shipped version of the targets shim changes which will happen in VS 15.8.

Pilchie commented on setup/resources/Microsoft.Portable.FSharp.Shim.targets:
```diff
@@ -1,5 +1,6 @@
-  
+
+  <Import Condition="'$(FSharpCompilerPath)' != ''" Project="$(FSharpCompilerPath)\Microsoft.Portable.FSharp.Targets" />
+  <Import Condition="'$(FSharpCompilerPath)' == ''" Project="$(MSBuildProgramFiles32)\Microsoft SDKs\F#\10.1\Framework\v4.0\Microsoft.Portable.FSharp.Targets" />
```
But nothing will install to this path anymore, since we removed the MSI?

KevinRansom replied:
That is fixed in this PR:  https://github.com/Microsoft/visualfsharp/pull/4819

Here:https://github.com/Microsoft/visualfsharp/pull/4819/files#diff-3c8a247250b0ee95f17ac1a0253f962fL1

brettfo replied:
I'll have to rebase that PR to fix this bottom line, but this will still work.  The MSI removal doesn't remove these files, they're just placed under `Common7\IDE\...`.

Pilchie commented:
I guess my overall point is that I'd prefer us to match C# as closely as we can for this stuff (keeping in mind that `FSharp.Core` throws a small wrench here).  I acknowledge that we still need a good way of having the F5'd vsix use the new compiler, but I'd prefer the shipped product mirror C#.

0x53A commented:
``FSharp.Compiler.Tools`` overrides the compiler using this property and mirrors ``Microsoft.Net.Compilers``.

Please continue to support both.

brettfo commented:
@0x53A Thanks for the pointer.  Looking at how [`FSharp.Compiler.Tools.props`](https://github.com/fsharp/fsharp/blob/76d5288b0b54e20b14a1bf5ca8eb0e9193c5485c/FSharp.Compiler.Tools.Nuget/build/FSharp.Compiler.Tools.props) works it should be good, but I'll modify my local install of VS to match what this PR does and verify just to be sure.

brettfo commented:
@0x53A It works as expected.  I did the following:

1.  Built these changes locally and deployed the VSIX (to ensure I have the code that sets `FSharpCompilerPath` on project load).
2.  Modified my local targets files to have the redirect that exists in this PR.
3.  Built from VS -> fsc was invoked from the deployed VSIX location.
4.  I added a reference to `FSharp.Compiler.Tools` version 10.0.2.
5.  Re-built from VS -> fsc was invoked from `..\packages\FSharp.Compiler.Tools\...`
6.  Re-built from the command line -> same as in step 5.
